src/Mailgun/Api/AbstractApi.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Mailgun\Api;
+
+use Mailgun\Exception\HttpServerException;
+use Mailgun\HttpClient\ResponseMediator;
+use Mailgun\Mailgun;
+use Http\Client\Exception as HttplugException;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Contributors of https://github.com/KnpLabs/php-github-api
+ */
+abstract class AbstractApi
+{
+    /**
+     * The client.
+     *
+     * @var Mailgun
+     */
+    protected $mailgun;
+
+    /**
+     * @param Mailgun $client
+     */
+    public function __construct(Mailgun $mailgun)
+    {
+        $this->mailgun = $mailgun;
+    }
+
+    /**
+     * Send a GET request with query parameters.
+     *
+     * @param string $path           Request path.
+     * @param array  $parameters     GET parameters.
+     * @param array  $requestHeaders Request Headers.
+     *
+     * @return array|string
+     */
+    protected function get($path, array $parameters = array(), $requestHeaders = array())
+    {
+        if (count($parameters) > 0) {
+            $path .= '?'.http_build_query($parameters);
+        }
+
+        try {
+            $response = $this->mailgun->getHttpClient()->get($path, $requestHeaders);
+        } catch (HttplugException\NetworkException $e) {
+            throw HttpServerException::networkError($e);
+        }
+
+        return ResponseMediator::getContent($response);
+    }
+
+
```

Comment:
Minor update - Code Styling:
Can you remove one of these lines please? We should provide just one empty line between two methods.